### PR TITLE
2016Q4.feature.containerlistenerfork

### PR DESCRIPTION
**Pull request by chrberger**
Signed-off-by: Christian Berger <christian.berger@gu.se>

### CODE REVIEW

**Comment by chrberger** [comment]
* Added ContainerListenerFork feature

File: libopendavinci/testsuites/ContainerTestSuite.h [hunk 1]
```diff
@@ -143,6 +162,79 @@ class ContainerTest : public CxxTest::TestSuite {
...
+            TS_ASSERT(!listener2.m_hasReceived);
+
+            clf.addContainerListener(&listener1);
+            TS_ASSERT(!listener1.m_hasReceived);
+            TS_ASSERT(!listener3.m_hasReceived);
+
+            clf.addContainerListener(&listener2);
+            TS_ASSERT(!listener1.m_hasReceived);
+            TS_ASSERT(!listener2.m_hasReceived);
+            TS_ASSERT(!listener3.m_hasReceived);
```
Comment (HangYin2015) [inline libopendavinci/testsuites/ContainerTestSuite.h:183]: What is the reason for asserting listeners in this order: first listener 1 and 2, then 1 and 3, and then 1-3?

File: libopendavinci/testsuites/ContainerTestSuite.h [hunk 2]
```diff
@@ -143,6 +162,79 @@ class ContainerTest : public CxxTest::TestSuite {
...
+            TS_ASSERT(listener2.m_hasReceived);
+            TS_ASSERT(!listener3.m_hasReceived);
+
+            TS_ASSERT(listener1.m_container.getDataType() == ts.ID());
+            TS_ASSERT(listener1.m_container.getData<TimeStamp>().toMicroseconds() == ts.toMicroseconds());
+
+            TS_ASSERT(listener2.m_container.getDataType() == ts.ID());
+            TS_ASSERT(listener2.m_container.getData<TimeStamp>().toMicroseconds() == ts.toMicroseconds());
+
+            clf.removeContainerListener(&listener3);
```
Comment (HangYin2015) [inline libopendavinci/testsuites/ContainerTestSuite.h:199]: I do not find where listener3 is added.

File: libopendavinci/testsuites/ContainerTestSuite.h (hunk 1 above)
Comment (chrberger) [reply libopendavinci/testsuites/ContainerTestSuite.h:183]: The motivation is to ensure that the listeners are not modified after adding to the ContainerListenersFork instance.

File: libopendavinci/testsuites/ContainerTestSuite.h (hunk 2 above)
Comment (chrberger) [reply libopendavinci/testsuites/ContainerTestSuite.h:199]: On purpose, listener3 will not be added.

**Review by HangYin2015 — APPROVED** [review]

**Pull request merged**